cmd/kubeapps-apis/plugins/fluxv2/packages/v1alpha1/repo_auth.go

```diff
@@ -0,0 +1,523 @@
+		return nil, false, false, status.Errorf(codes.InvalidArgument, "Package repository cannot mix referenced secrets and user provided secret data")
+	}
+
+	typedClient, err := s.clientGetter.Typed(ctx, s.kubeappsCluster)
+	if err != nil {
+		return nil, false, false, err
+	}
+	secretInterface := typedClient.CoreV1().Secrets(repo.Namespace)
+
+	var existingSecret *apiv1.Secret
+	if repo.Spec.SecretRef != nil {
+		if existingSecret, err = secretInterface.Get(ctx, repo.Spec.SecretRef.Name, metav1.GetOptions{}); err != nil {
+			return nil, false, false, statuserror.FromK8sError("get", "secret", repo.Spec.SecretRef.Name, err)
+		}
+	}
+
+	// check we cannot change mode (per design spec)
+	if existingSecret != nil && (hasCaRef || hasCaData || hasAuthRef || hasAuthData) {
+		if isSecretKubeappsManaged(existingSecret, repo) != (hasAuthData || hasCaData) {
+			return nil, false, false, status.Errorf(codes.InvalidArgument, "Auth management mode cannot be changed")
+		}
+	}
+
+	repoName := types.NamespacedName{Name: repo.Name, Namespace: repo.Namespace}
+	repoType := repo.Spec.Type
+
+	// handle user managed secret
+	if hasCaRef || hasAuthRef {
+		updatedSecret, err := s.validateUserManagedRepoSecret(ctx, repoName, repoType, newTlsConfig, newAuth)
+		return updatedSecret, false, true, err
+	}
+
+	// handle kubeapps managed secret
+	var isSameSecret bool
+	updatedSecret, isSameSecret, err = newSecretFromTlsConfigAndAuth(repoName, repoType, newTlsConfig, newAuth)
+	if err != nil {
+		return nil, true, false, err
+	} else if isSameSecret {
+		// Do nothing if repo auth data came redacted
+		return nil, true, false, nil
+	} else {
+		// either we have no secret, or it has changed. in both cases, we try to delete any existing secret
+		if existingSecret != nil {
+			if err = secretInterface.Delete(ctx, existingSecret.Name, metav1.DeleteOptions{}); err != nil {
+				log.Errorf("Error deleting existing secret: [%s] due to %v", err)
+			}
+		}
+		if updatedSecret != nil {
+			if updatedSecret, err = typedClient.CoreV1().Secrets(repoName.Namespace).Create(ctx, updatedSecret, metav1.CreateOptions{}); err != nil {
+				return nil, false, false, statuserror.FromK8sError("create", "secret", updatedSecret.GetGenerateName(), err)
+			}
+		}
+		return updatedSecret, true, true, nil
+	}
+}
+
+func (s *Server) validateUserManagedRepoSecret(
+	ctx context.Context,
+	repoName types.NamespacedName,
+	repoType string,
+	tlsConfig *corev1.PackageRepositoryTlsConfig,
+	auth *corev1.PackageRepositoryAuth) (*apiv1.Secret, error) {
+	var secretRefTls, secretRefAuth string
+	if tlsConfig != nil {
+		if tlsConfig.GetCertAuthority() != "" {
+			return nil, status.Errorf(codes.InvalidArgument, "Secret Ref must be used with user managed secrets")
+		} else if tlsConfig.GetSecretRef().GetName() != "" {
+			secretRefTls = tlsConfig.GetSecretRef().GetName()
+		}
+	}
+
+	if auth != nil {
+		if auth.GetDockerCreds() != nil ||
+			auth.GetHeader() != "" ||
```

Comment:
is "header" a auth type supported by flux?

Reply:
nope. 
https://github.com/vmware-tanzu/kubeapps/blob/f5b14866554aa5464c699938e5cf9a9231869281/cmd/kubeapps-apis/plugins/fluxv2/packages/v1alpha1/repo.go#L1244
This particular piece of code has not changed in this PR.